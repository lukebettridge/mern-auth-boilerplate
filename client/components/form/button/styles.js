import styled, { css } from "styled-components";

const Button = styled.button`
	background-color: #f8f8f8;
	border: 1px solid #eeeeee;
	cursor: pointer;
	font-size: 13pt;
	font-weight: 600;
	max-width: 350px;
	padding: 8px 20px;
	width: 100%;

	${props =>
		props.as === "a" &&
		css`
			background-color: transparent;
			color: #596f86;
			display: block;
			font-weight: 400;
			max-width: none;
			text-decoration: none;
			&:hover {
				text-decoration: underline;
			}

			${props.secondary &&
				css`
					color: #aaaaaa;
					font-size: 11pt;
					font-weight: 300;
				`}
		`}

	${props =>
		props.mt === "s" &&
		css`
			margin-top: 16px;
		`}

	${props =>
		props.mb === "s" &&
		css`
			margin-bottom: 16px;
		`}
`;

module.exports = { Button };
