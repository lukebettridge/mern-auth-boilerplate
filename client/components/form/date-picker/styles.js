import styled, { css } from "styled-components";

const DatePickerContainer = styled.div`
	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}

	${props =>
		(props.value || props.focused) &&
		css`
			& + label {
				font-size: 95%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}
`;

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Container = styled.div`
	position: relative;

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}

	.SingleDatePicker,
	.SingleDatePickerInput,
	.DateInput {
		display: block;
		width: 100%;
	}

	.DateInput_input {
		background-color: transparent;
		border-bottom: 1px solid #e9ecef;
		font-size: 11pt;
		font-weight: 300;
		letter-spacing: 1px;
		margin-bottom: 8px;
		padding: 12px 10px 5px;
		transition: border-color 100ms;
		width: 100%;

		&:hover {
			border-color: #b3b3b3;
		}

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}
	}
`;

const Label = styled.label`
	color: #999999;
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Container, DatePickerContainer, Error, Label };
