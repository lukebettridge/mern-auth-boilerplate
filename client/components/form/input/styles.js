import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Input = styled.input`
	background-color: transparent;
	border: 0;
	border-bottom: 1px solid #e9ecef;
	box-shadow: none;
	font-size: 11pt;
	font-weight: 300;
	margin-bottom: 8px;
	outline: none;
	padding: 12px 10px;
	width: 100%;

	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			border-color: #db1802;

			& + label {
				color: #db1802;
			}
		`}
`;

const InputContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 20px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Input, InputContainer, Label };
