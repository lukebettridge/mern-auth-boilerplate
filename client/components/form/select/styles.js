import styled, { css } from "styled-components";
import breakpoints from "components/styles/breakpoints";
import { default as ReactSelect } from "react-select";

const Error = styled.p`
	color: #db1802;
	font-size: 10pt;
	font-weight: 300;
	margin-bottom: 16px;
	margin-top: 0;
	&:first-letter {
		text-transform: capitalize;
	}
`;

const Select = styled(ReactSelect)`
	&:focus + label {
		font-size: 75%;
		transform: translate3d(0, -100%, 0);
		opacity: 1;
	}

	${props =>
		props.value &&
		props.value.length > 0 &&
		css`
			& + label {
				font-size: 75%;
				transform: translate3d(0, -100%, 0);
				opacity: 1;
			}
		`}

	${props =>
		props.inError &&
		css`
			& + label {
				color: #db1802;
			}
		`}
        
	.react-select__control {
		appearance: none;
		background-color: transparent;
		border: 0;
		border-bottom: 1px solid #e9ecef;
		border-radius: 0;
		box-shadow: none;
		font-size: 12pt;
		font-weight: 300;
		height: 42px;
		margin-bottom: 8px;
		outline: none;
		width: 100%;

		${props =>
			props.inError &&
			css`
				border-color: #db1802;
			`}

		.react-select__input {
			opacity: 0;
		}
	}
`;

const SelectContainer = styled.div`
	position: relative;
	width: 100%;

	@media (min-width: ${breakpoints.m}) {
		min-width: 250px;
	}

	${props =>
		props.label &&
		css`
			margin-top: 30px;
		`}

	${props =>
		props.mb &&
		css`
			margin-bottom: 30px;
		`}
`;

const Label = styled.label`
	left: 0;
	opacity: 0.5;
	font-weight: 300;
	padding-left: 10px;
	padding-top: 10px;
	position: absolute;
	top: 0;
	transition: all 200ms;
`;

export { Error, Select, SelectContainer, Label };
