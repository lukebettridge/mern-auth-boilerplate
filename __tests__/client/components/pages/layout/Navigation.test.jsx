import React from "react";
import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import cloneDeep from "lodash/cloneDeep";
import { MemoryRouter } from "react-router-dom";

import * as S from "components/pages/layout/styles";

import Navigation from "components/pages/layout/Navigation";

let wrapper;
let subject;

describe("Layout component", () => {
	beforeEach(() => {
		jest.resetAllMocks();
	});

	afterEach(() => {
		if (wrapper) wrapper.unmount();
	});

	it("snapshot renders", () => {
		mountWrapper();
		expect(toJSON(subject)).toMatchSnapshot();
	});

	it("renders component", () => {
		mountWrapper();

		expect(subject).not.toBeNull();
		expect(subject.find(S.Anchor).html()).toContain("John");
		expect(subject.find(S.ListItem).length).toEqual(2);
	});

	it("set anchor to active", () => {
		mountWrapper();

		subject.find(S.Anchor).simulate("click");
		updateSubject();

		expect(subject.find(S.Anchor).props().active).toEqual(true);
	});

	it("show account page for admin", () => {
		const props = cloneDeep(baseProps);
		Object.assign(props.currentUser, { roles: ["admin"] });

		mountWrapper(props);

		expect(subject.find(S.ListItem).length).toEqual(3);
		expect(
			subject
				.find(S.ListItem)
				.at(2)
				.html()
		).toContain("Accounts");
	});

	it("calls edit profile prop", () => {
		mountWrapper();

		subject
			.find(S.SubAnchor)
			.at(0)
			.simulate("click");

		expect(baseProps.editProfile).toHaveBeenCalled();
	});

	it("calls change password prop", () => {
		mountWrapper();

		subject
			.find(S.SubAnchor)
			.at(1)
			.simulate("click");

		expect(baseProps.changePassword).toHaveBeenCalled();
	});
});

const mountWrapper = (props = baseProps) => {
	wrapper = mount(
		<MemoryRouter initialEntries={[{ pathname: "/path", key: "key" }]}>
			<Navigation {...props} />
		</MemoryRouter>
	);
	subject = wrapper.find(Navigation);
};

const updateSubject = () => {
	wrapper.update();
	subject = wrapper.find(Navigation);
};

const baseProps = {
	changePassword: jest.fn(),
	currentUser: {
		forename: "John",
		roles: []
	},
	editProfile: jest.fn()
};
