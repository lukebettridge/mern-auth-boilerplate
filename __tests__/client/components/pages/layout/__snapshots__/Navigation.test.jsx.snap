// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component snapshot renders 1`] = `
.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #294661;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12pt;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 0.6;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c3:hover {
  opacity: 1;
}

.c3 span svg {
  font-size: 18pt;
  margin-right: 6px;
  padding-bottom: 3px;
  vertical-align: middle;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #294661;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12pt;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 0.6;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
}

.c8:hover {
  opacity: 1;
}

.c8 span svg {
  font-size: 18pt;
  margin-right: 6px;
  padding-bottom: 3px;
  vertical-align: middle;
}

.c1 {
  margin: 0;
}

.c2 {
  margin-bottom: 40px;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #294661;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12pt;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 0.6;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  font-size: 11pt;
  padding-left: 15px;
}

.c6:hover {
  opacity: 1;
}

.c6 span svg {
  font-size: 18pt;
  margin-right: 6px;
  padding-bottom: 3px;
  vertical-align: middle;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #294661;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12pt;
  font-weight: 300;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  opacity: 0.6;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 0.3s;
  transition: opacity 0.3s;
  font-size: 11pt;
  padding-left: 15px;
}

.c7:hover {
  opacity: 1;
}

.c7 span svg {
  font-size: 18pt;
  margin-right: 6px;
  padding-bottom: 3px;
  vertical-align: middle;
}

.c4 {
  margin-top: 15px;
  max-height: 1000px;
  overflow: hidden;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  margin: 0;
  max-height: 0;
}

.c5 {
  margin-bottom: 8px;
}

.c0 {
  background-color: #fbfbfc;
  border-right: 1px solid #e9ecef;
  display: none;
  height: 100%;
  max-width: 200px;
  min-width: 200px;
  padding-left: 15px;
  padding-right: 10px;
  padding-top: 48px;
  width: 200px;
  z-index: 1;
}

@media (min-width:768px) {
  .c0 {
    display: table-cell;
    height: unset;
    position: unset;
  }
}

<Navigation
  changePassword={[MockFunction]}
  currentUser={
    Object {
      "forename": "John",
      "roles": Array [],
    }
  }
  editProfile={[MockFunction]}
>
  <styles__Navigation
    changePassword={[MockFunction]}
    currentUser={
      Object {
        "forename": "John",
        "roles": Array [],
      }
    }
    editProfile={[MockFunction]}
  >
    <StyledComponent
      changePassword={[MockFunction]}
      currentUser={
        Object {
          "forename": "John",
          "roles": Array [],
        }
      }
      editProfile={[MockFunction]}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "styles__Navigation-sc-1fs86t6-11",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "background-color:#fbfbfc;border-right:1px solid #e9ecef;display:none;height:100%;max-width:200px;min-width:200px;padding-left:15px;padding-right:10px;padding-top:48px;width:200px;z-index:1;",
              [Function],
              " @media (min-width:",
              "768px",
              "){display:table-cell;height:unset;position:unset;}",
            ],
          },
          "displayName": "styles__Navigation",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "styles__Navigation-sc-1fs86t6-11",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styles__List>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "styles__List-sc-1fs86t6-4",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "margin:0;",
                  ],
                },
                "displayName": "styles__List",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "styles__List-sc-1fs86t6-4",
                "target": "ul",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <ul
              className="c1"
            >
              <styles__ListItem>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__ListItem-sc-1fs86t6-5",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "margin-bottom:40px;",
                        ],
                      },
                      "displayName": "styles__ListItem",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__ListItem-sc-1fs86t6-5",
                      "target": "li",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <li
                    className="c2"
                  >
                    <styles__Anchor
                      active={false}
                      onClick={[Function]}
                    >
                      <StyledComponent
                        active={false}
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__Anchor-sc-1fs86t6-0",
                              "isStatic": false,
                              "lastClassName": "c3",
                              "rules": Array [
                                "
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}
",
                                " ",
                                [Function],
                              ],
                            },
                            "displayName": "styles__Anchor",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "styles__Anchor-sc-1fs86t6-0",
                            "target": "a",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        onClick={[Function]}
                      >
                        <a
                          className="c3"
                          onClick={[Function]}
                        >
                          <span
                            data-cy="user"
                          >
                            <FiUser>
                              <IconBase
                                attr={
                                  Object {
                                    "fill": "none",
                                    "stroke": "currentColor",
                                    "strokeLinecap": "round",
                                    "strokeLinejoin": "round",
                                    "strokeWidth": "2",
                                    "viewBox": "0 0 24 24",
                                  }
                                }
                              >
                                <svg
                                  fill="none"
                                  height="1em"
                                  stroke="currentColor"
                                  strokeLinecap="round"
                                  strokeLinejoin="round"
                                  strokeWidth="2"
                                  style={
                                    Object {
                                      "color": undefined,
                                    }
                                  }
                                  viewBox="0 0 24 24"
                                  width="1em"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
                                    key="0"
                                  />
                                  <circle
                                    cx="12"
                                    cy="7"
                                    key="1"
                                    r="4"
                                  />
                                </svg>
                              </IconBase>
                            </FiUser>
                            John
                          </span>
                          <FiChevronDown>
                            <IconBase
                              attr={
                                Object {
                                  "fill": "none",
                                  "stroke": "currentColor",
                                  "strokeLinecap": "round",
                                  "strokeLinejoin": "round",
                                  "strokeWidth": "2",
                                  "viewBox": "0 0 24 24",
                                }
                              }
                            >
                              <svg
                                fill="none"
                                height="1em"
                                stroke="currentColor"
                                strokeLinecap="round"
                                strokeLinejoin="round"
                                strokeWidth="2"
                                style={
                                  Object {
                                    "color": undefined,
                                  }
                                }
                                viewBox="0 0 24 24"
                                width="1em"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <polyline
                                  key="0"
                                  points="6 9 12 15 18 9"
                                />
                              </svg>
                            </IconBase>
                          </FiChevronDown>
                        </a>
                      </StyledComponent>
                    </styles__Anchor>
                    <styles__SubList
                      active={false}
                    >
                      <StyledComponent
                        active={false}
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__SubList-sc-1fs86t6-9",
                              "isStatic": false,
                              "lastClassName": "c4",
                              "rules": Array [
                                "margin-top:15px;max-height:1000px;overflow:hidden;transition:all 0.3s;",
                                [Function],
                              ],
                            },
                            "displayName": "styles__SubList",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "styles__SubList-sc-1fs86t6-9",
                            "target": "ul",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <ul
                          className="c4"
                        >
                          <styles__SubListItem>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__SubListItem-sc-1fs86t6-10",
                                    "isStatic": false,
                                    "lastClassName": "c5",
                                    "rules": Array [
                                      "margin-bottom:8px;",
                                    ],
                                  },
                                  "displayName": "styles__SubListItem",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__SubListItem-sc-1fs86t6-10",
                                  "target": "li",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <li
                                className="c5"
                              >
                                <styles__SubAnchor
                                  onClick={[MockFunction]}
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "styles__SubAnchor-sc-1fs86t6-7",
                                          "isStatic": false,
                                          "lastClassName": "c6",
                                          "rules": Array [
                                            "
	
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}

	font-size: 11pt;
	padding-left: 15px;
",
                                          ],
                                        },
                                        "displayName": "styles__SubAnchor",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "styles__SubAnchor-sc-1fs86t6-7",
                                        "target": "a",
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    onClick={[MockFunction]}
                                  >
                                    <a
                                      className="c6"
                                      onClick={[MockFunction]}
                                    >
                                      Edit Profile
                                    </a>
                                  </StyledComponent>
                                </styles__SubAnchor>
                              </li>
                            </StyledComponent>
                          </styles__SubListItem>
                          <styles__SubListItem>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__SubListItem-sc-1fs86t6-10",
                                    "isStatic": false,
                                    "lastClassName": "c5",
                                    "rules": Array [
                                      "margin-bottom:8px;",
                                    ],
                                  },
                                  "displayName": "styles__SubListItem",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__SubListItem-sc-1fs86t6-10",
                                  "target": "li",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <li
                                className="c5"
                              >
                                <styles__SubAnchor
                                  onClick={[MockFunction]}
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "styles__SubAnchor-sc-1fs86t6-7",
                                          "isStatic": false,
                                          "lastClassName": "c6",
                                          "rules": Array [
                                            "
	
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}

	font-size: 11pt;
	padding-left: 15px;
",
                                          ],
                                        },
                                        "displayName": "styles__SubAnchor",
                                        "foldedComponentIds": Array [],
                                        "render": [Function],
                                        "styledComponentId": "styles__SubAnchor-sc-1fs86t6-7",
                                        "target": "a",
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    onClick={[MockFunction]}
                                  >
                                    <a
                                      className="c6"
                                      onClick={[MockFunction]}
                                    >
                                      Change Password
                                    </a>
                                  </StyledComponent>
                                </styles__SubAnchor>
                              </li>
                            </StyledComponent>
                          </styles__SubListItem>
                          <styles__SubListItem>
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "styles__SubListItem-sc-1fs86t6-10",
                                    "isStatic": false,
                                    "lastClassName": "c5",
                                    "rules": Array [
                                      "margin-bottom:8px;",
                                    ],
                                  },
                                  "displayName": "styles__SubListItem",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "styles__SubListItem-sc-1fs86t6-10",
                                  "target": "li",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={null}
                            >
                              <li
                                className="c5"
                              >
                                <styles__SubLink
                                  to="/auth/logout"
                                >
                                  <StyledComponent
                                    forwardedComponent={
                                      Object {
                                        "$$typeof": Symbol(react.forward_ref),
                                        "attrs": Array [],
                                        "componentStyle": ComponentStyle {
                                          "componentId": "styles__SubLink-sc-1fs86t6-8",
                                          "isStatic": false,
                                          "lastClassName": "c7",
                                          "rules": Array [
                                            "
	
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}

	font-size: 11pt;
	padding-left: 15px;
",
                                          ],
                                        },
                                        "displayName": "styles__SubLink",
                                        "foldedComponentIds": Array [],
                                        "propTypes": Object {
                                          "innerRef": [Function],
                                          "onClick": [Function],
                                          "replace": [Function],
                                          "target": [Function],
                                          "to": [Function],
                                        },
                                        "render": [Function],
                                        "styledComponentId": "styles__SubLink-sc-1fs86t6-8",
                                        "target": Object {
                                          "$$typeof": Symbol(react.forward_ref),
                                          "displayName": "Link",
                                          "propTypes": Object {
                                            "innerRef": [Function],
                                            "onClick": [Function],
                                            "replace": [Function],
                                            "target": [Function],
                                            "to": [Function],
                                          },
                                          "render": [Function],
                                        },
                                        "toString": [Function],
                                        "warnTooManyClasses": [Function],
                                        "withComponent": [Function],
                                      }
                                    }
                                    forwardedRef={null}
                                    to="/auth/logout"
                                  >
                                    <Link
                                      className="c7"
                                      to="/auth/logout"
                                    >
                                      <LinkAnchor
                                        className="c7"
                                        href="/auth/logout"
                                        navigate={[Function]}
                                      >
                                        <a
                                          className="c7"
                                          href="/auth/logout"
                                          onClick={[Function]}
                                        >
                                          Logout
                                        </a>
                                      </LinkAnchor>
                                    </Link>
                                  </StyledComponent>
                                </styles__SubLink>
                              </li>
                            </StyledComponent>
                          </styles__SubListItem>
                        </ul>
                      </StyledComponent>
                    </styles__SubList>
                  </li>
                </StyledComponent>
              </styles__ListItem>
              <styles__ListItem>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "styles__ListItem-sc-1fs86t6-5",
                        "isStatic": false,
                        "lastClassName": "c2",
                        "rules": Array [
                          "margin-bottom:40px;",
                        ],
                      },
                      "displayName": "styles__ListItem",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "styles__ListItem-sc-1fs86t6-5",
                      "target": "li",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <li
                    className="c2"
                  >
                    <styles__Link
                      to="/"
                    >
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "styles__Link-sc-1fs86t6-3",
                              "isStatic": false,
                              "lastClassName": "c8",
                              "rules": Array [
                                "
	align-items: center;
	color: #294661;
	cursor: pointer;
	display: flex;
	font-size: 12pt;
	font-weight: 300;
	justify-content: space-between;
	opacity: 0.6;
	text-decoration: none;
	transition: opacity 0.3s;

	&:hover {
		opacity: 1;
	}

	span svg {
		font-size: 18pt;
		margin-right: 6px;
		padding-bottom: 3px;
		vertical-align: middle;
	}
",
                              ],
                            },
                            "displayName": "styles__Link",
                            "foldedComponentIds": Array [],
                            "propTypes": Object {
                              "innerRef": [Function],
                              "onClick": [Function],
                              "replace": [Function],
                              "target": [Function],
                              "to": [Function],
                            },
                            "render": [Function],
                            "styledComponentId": "styles__Link-sc-1fs86t6-3",
                            "target": Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "Link",
                              "propTypes": Object {
                                "innerRef": [Function],
                                "onClick": [Function],
                                "replace": [Function],
                                "target": [Function],
                                "to": [Function],
                              },
                              "render": [Function],
                            },
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                        to="/"
                      >
                        <Link
                          className="c8"
                          to="/"
                        >
                          <LinkAnchor
                            className="c8"
                            href="/"
                            navigate={[Function]}
                          >
                            <a
                              className="c8"
                              href="/"
                              onClick={[Function]}
                            >
                              <span>
                                <FiCheckCircle>
                                  <IconBase
                                    attr={
                                      Object {
                                        "fill": "none",
                                        "stroke": "currentColor",
                                        "strokeLinecap": "round",
                                        "strokeLinejoin": "round",
                                        "strokeWidth": "2",
                                        "viewBox": "0 0 24 24",
                                      }
                                    }
                                  >
                                    <svg
                                      fill="none"
                                      height="1em"
                                      stroke="currentColor"
                                      strokeLinecap="round"
                                      strokeLinejoin="round"
                                      strokeWidth="2"
                                      style={
                                        Object {
                                          "color": undefined,
                                        }
                                      }
                                      viewBox="0 0 24 24"
                                      width="1em"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
                                        key="0"
                                      />
                                      <polyline
                                        key="1"
                                        points="22 4 12 14.01 9 11.01"
                                      />
                                    </svg>
                                  </IconBase>
                                </FiCheckCircle>
                                Todos
                              </span>
                            </a>
                          </LinkAnchor>
                        </Link>
                      </StyledComponent>
                    </styles__Link>
                  </li>
                </StyledComponent>
              </styles__ListItem>
            </ul>
          </StyledComponent>
        </styles__List>
      </div>
    </StyledComponent>
  </styles__Navigation>
</Navigation>
`;
